src/org/labkey/test/tests/TriggerScriptTest.java

```diff
@@ -679,23 +679,29 @@ private void insertSingleRowViaUI(String dataRegionName, Map<String,String> reco
 
     /**
      * delete single record via the table UI
-     * @param columnName Column to look at
-     * @param columnValue value to look for
-     * @param tableName DataRegionTable name
+     *
+     * @param columnName       Column to look at
+     * @param columnValue      value to look for
+     * @param tableName        DataRegionTable name
+     * @param deleteButtonText text that appears in delete confirmation when not in an alert.
+     * @param expectPageLoad indicates whether confirming deletion will result in a page load or not
      */
-    private void deleteSingleRowViaUI(String columnName, String columnValue, String tableName)
+    private void deleteSingleRowViaUI(String columnName, String columnValue, String tableName, String deleteButtonText, boolean expectPageLoad)
     {
         DataRegionTable drt = new DataRegionTable(tableName, this);
         int rowId = drt.getRowIndex(columnName, columnValue);
         drt.checkCheckbox(rowId);
-        doAndWaitForPageToLoad(() ->
+        doAndMaybeWaitForPageToLoad(1000, () ->
         {
             drt.clickHeaderButton("Delete");
             Alert alert = getAlertIfPresent();
             if (alert != null)
                 alert.accept();
+            else if (expectPageLoad)
+                clickButton(deleteButtonText);
             else
-                clickButton("Confirm Delete");
+                click(Locator.linkWithText(deleteButtonText));
```

Comment:
I don't think this needs to check the `expectPageLoad` flag. The surrounding `doAndMaybeWaitForPageToLoad` will handle the page load if necessary. You can pass a zero timeout to `clickButton` to suppress its page load expectation.
```
if (alert != null)
    alert.accept();
else
    clickButton(deleteButtonText, 0);
```

Reply:
Done.